case_studies/CARD/Bug-Drug Code.R

```diff
@@ -0,0 +1,257 @@
+#Install and Load required packages
+if (!requireNamespace("rentrez", quietly = TRUE)) {
+  install.packages("rentrez")
+}
+if (!requireNamespace("XML", quietly = TRUE)) {
+  install.packages("XML")
+}
+if (!requireNamespace("stringr", quietly = TRUE)) {
+  install.packages("stringr")
+}
+
+
+library(rentrez)
+library(XML)
+library(stringr)
+
+# Filter for the target drug (DAP) and pathogen (Staphylococcus aureus)
+filter_resistance_mechanisms <- function(data, drug, bug, exclude_multiclass = FALSE, species_restricted = TRUE) {
+
+  # Filter by drug using partial match to include multiclass entries containing the target drug
+  filtered_data <- data %>%
+    filter(grepl(drug, Drug, ignore.case = TRUE))
+
+  # Filter by pathogen, using partial match
+  filtered_data <- filtered_data %>%
+    filter(grepl(bug, Pathogen_Full_Name, ignore.case = TRUE))
+
+  # Optionally exclude multiclass resistance if exclude_multiclass = TRUE
+  if (exclude_multiclass) {
+    filtered_data <- filtered_data %>%
+      filter(!grepl(";", Drug))  # Only include entries with single drug classes
+  }
+
+  # Optionally restrict to species-specific mechanisms if species_restricted = TRUE
+  if (species_restricted) {
+    filtered_data <- filtered_data %>%
+      filter(Pathogen_Full_Name == bug)  # Include only entries with exact match to the bug of interest
+  }
+
+  return(filtered_data)
+}
+
+# Usage example for Staphylococcus aureus resistant to DAP, including multispecies and multiclass resistance
+filtered_data_saurdap <- filter_resistance_mechanisms(
+  data = merged_data_pathogens,
+  drug = "DAP",
+  bug = "Staphylococcus aureus",
+  exclude_multiclass = FALSE,
+  species_restricted = FALSE
+)
+
+# View the filtered results
+View(filtered_data_saurdap)
+
+
+# Fetch FASTA sequence from Entrez
+fetch_fasta_sequence <- function(protein_accession) {
+  tryCatch({
+    # Fetch the FASTA sequence using Entrez
+    fasta_seq <- rentrez::entrez_fetch(db = "protein",
+                                       id = protein_accession,
+                                       rettype = "fasta",
+                                       retmode = "text")
+    
+    if (!is.null(fasta_seq)) {
+      # Ensure the first line starts with ">"
+      if (!grepl("^>", fasta_seq[1])) {
+        fasta_seq[1] <- paste0(">", fasta_seq[1])
+      }
+      
+      # Split the sequence into lines
+      lines <- str_split(fasta_seq, "\n")[[1]]
+      
+      # Join the lines back together
+      fasta_seq <- paste(lines, collapse = "\n")
+      
+      return(fasta_seq)
+    } else {
+      warning(paste("Failed to retrieve FASTA sequence for protein accession:", protein_accession))
+      return(NULL)
+    }
+  }, error = function(e) {
+    warning(paste("Error fetching FASTA sequence for protein accession:", protein_accession, ":", e$message))
+    return(NULL)
+  })
+}
+
+
+# Define the output file for the FASTA sequences
+output_fasta_file <- "Staph_aureus_Daptomycin_sequences.fasta"
```

Comment:
If using short names for species (4-char) and drugs (antibiotics, 3-char).
arg = antibiotic resistance genes, for example.
Which shortnames are you planning to use?
cc: @AbhirupaGhosh @charmvang @awasyn @epbrenner 

```suggestion
output_fasta_file <- "Saur_dap_arg.fasta"
```